### PR TITLE
[WIP] Individual contributions map

### PR DESCRIPTION
**Pull request by noahmanger**
This creates the map of individual contributions for the raising breakdown page. So far this is just the start of it.

Currently just uses data from a single committee endpoint to test it, and also just uses the established styles from these maps used on committee pages:

image

### CODE REVIEW

**Comment by xtine** [comment]
Full width layout, live data (with filtering):
screen shot 2016-11-08 at 2 15 27 pm

**Comment by xtine** [comment]
Need these:
- Methodology data
- What the "Data" icon links to
- Variable for reaction chart ("What do you think of this chart") for map

File: openfecwebapp/templates/layouts/sidebar-page.html [hunk 1]
```diff
@@ -11,12 +11,12 @@
 {{ header.header(title, '', show_search=False) }}
 
 <div class="main">
-  <div class="container">
+  <div class="breakdown-container">
```
Comment (noahmanger) [inline openfecwebapp/templates/layouts/sidebar-page.html]: Can we change this to something more generic. Maybe `container--full-width`?

File: openfecwebapp/templates/raising-breakdown.html [hunk 2]
```diff
@@ -20,18 +21,70 @@ <h1 class="icon-heading--raising-circle">{{ title }}</h1>
...
     <h2>Who's raising the most</h2>
     {{ breakdowns.top_entities(top_raisers, coverage_start_date, coverage_end_date, cycle, 'raised', top_category, pagination ) }}
   </section>
+
+  <section id="by-state" class="content__section--extra content__section--ruled js-breakdown-map" data-cycle="{{ cycle }}">
+    <h2>Where contributions come from</h2>
+    <form action="" method="GET">
+      <h3 class="u-no-margin">
+        <label for="top-category" class="breakdown__title label t-inline-block">A state-by-state breakdown of individual contributions from each state to </label>
+        <select id="top-category" name="top_category" class="js-category form-element--inline" aria-controls="top-table">
```
Comment (noahmanger) [inline openfecwebapp/templates/raising-breakdown.html]: This needs to have a different ID, because this is shared by the top list above.

Comment (noahmanger) [inline openfecwebapp/templates/raising-breakdown.html]: (This is my bad. I didn't catch it when I copied it earlier).

File: openfecwebapp/templates/spending-breakdown.html [hunk 3]
```diff
@@ -32,7 +33,59 @@ <h1 class="icon-heading--spending-circle">{{ title }}</h1>
     <h2>Who's spending the most</h2>
     {{ breakdowns.top_entities(top_spenders, coverage_start_date, coverage_end_date, cycle, 'spent', top_category, page_info ) }}
   </section>
+
+{#   <section id="by-state" class="content__section--extra content__section--ruled js-breakdown-map" data-cycle="{{ cycle }}">
+    <h2>Where money is spent to support and oppose candidates</h2>
+    <form action="" method="GET">
+      <h3 class="u-no-margin">
+        <label for="top-category" class="breakdown__title label t-inline-block">A state-by-state breakdown of independent expenditures made by: </label>
+        <select id="top-category" name="top_category" class="js-category form-element--inline" aria-controls="top-table">
```
Comment (noahmanger) [inline openfecwebapp/templates/spending-breakdown.html]: This one will also need a unique ID

File: static/js/modules/breakdown-map.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+};
+
+BreakdownMap.prototype.destroyTableAndMap = function() {
+  this.stateTable.destroy();
+  this.$map.find('svg').remove();
+  this.$map.find('.legend-container').append('<svg></svg>');
+};
+
+BreakdownMap.prototype.updateDates = function() {
+  var today = new Date();
```
Comment (noahmanger) [inline static/js/modules/breakdown-map.js]: Is this code shared with the `TopEntities` component? If so maybe should be generalized into a helper method?

**Review by noahmanger — COMMENTED** [review]
Just caught a few small things, though I know there's still some bigger questions. Can you also add tests in this PR? You can look at the `top-entities-breakdown.js` test file for a model to copy.

**Comment by jenniferthibault** [comment]
Transcribed in`fec-cms` as a new issue to pick up this PR in the future when ready. Closing in favor of https://github.com/18F/fec-cms/issues/1662

**Pull request closed without merging**